src/dlgProfilePreferences.cpp

```diff
@@ -439,47 +440,159 @@ void dlgProfilePreferences::initWithHost(Host* pHost)
+    // Use the affix file as that may eliminate supplimental dictionaries:
+    QRegularExpression rex(QStringLiteral(R"(\.aff$)"));
     entries = entries.filter(rex);
-    for (int i = 0; i < entries.size(); i++) {
-        // This is a file name and to support macOs platforms should not be case sensitive:
-        entries[i].remove(QLatin1String(".dic"), Qt::CaseInsensitive);
-        auto item = new QListWidgetItem(entries[i]);
-        dictList->addItem(item);
-        if (entries[i] == pHost->mSpellDic) {
-            item->setSelected(true);
+    // Don't emit signals - like (void) QListWidget::currentItemChanged(QListWidgetItem *current, QListWidgetItem *previous)
+    // whilst populating the widget, it reduces noise about:
+    // "qt.accessibility.core: Cannot create accessible child interface for object:  QListWidget(0x############, name = "dictList")  index:  ##
+    dictList->blockSignals(true);
+    if (entries.count()) {
+        QListWidgetItem* scrollToItem = nullptr;
+        for (int i = 0, total = entries.size(); i < total; ++i) {
+            // This is a file name and to support macOs platforms should not be case sensitive:
+            entries[i].remove(QLatin1String(".aff"), Qt::CaseInsensitive);
+
+            if (entries.at(i).endsWith(QStringLiteral("med"), Qt::CaseInsensitive)) {
+                // Skip medical dictionaries - there may be others  we also want to hide:
+                continue;
+            }
+
+            auto item = new QListWidgetItem();
+            item->setTextAlignment(Qt::AlignCenter);
+            auto key = entries.at(i).toLower();
+            // In some cases '-' will be used as a separator and in others '_' so convert all to one form:
+            key.replace(QLatin1String("-"), QLatin1String("_"));
+            if (mudlet::self()->mDictionaryLanguageCodeMap.contains(key)) {
+                item->setText(mudlet::self()->mDictionaryLanguageCodeMap.value(key));
+                item->setToolTip(tr("<p>From the dictionary file <tt>%1.dic</tt> (and its companion affix <tt>.aff</tt> file).</p>").arg(dir.absoluteFilePath(entries.at(i))));
+            } else {
+                item->setText(tr("%1 - not recognised").arg(entries.at(i)));
+                item->setToolTip(tr("<p>Mudlet does not recognise the code \"%1\", please report it to the Mudlet developers so we can describe it properly in future Mudlet versions!</p>"
+                                    "<p>The file <tt>%2.dic</tt> (and its companion affix <tt>.aff</tt> file) is still usable.</p>").arg(entries.at(i), dir.absoluteFilePath(entries.at(i))));
+            }
+            item->setData(Qt::UserRole, entries.at(i));
+            dictList->addItem(item);
+            if (entries.at(i) == currentDictionary) {
+                scrollToItem = item;
+            }
         }
+
+        // Reenable sorting now we have populated the widget:
+        dictList->setSortingEnabled(false);
```

Comment:
```suggestion
        dictList->setSortingEnabled(true);
```

Reply:
:man_facepalming: Yeah! - Will fix that.

Reply:
In next commit.